libpod/container_internal_common.go

```diff
@@ -178,6 +178,51 @@ func getOverlayUpperAndWorkDir(options []string) (string, string, error) {
 	return upperDir, workDir, nil
 }
 
+// Internal only function which creates the Rootfs for default internal
+// pause image, configures the Rootfs in the Container and returns
+// the mount-point for the /catatonit. This mount-point should be added
+// to the Container spec.
+func (c *Container) prepareInitRootfs() (spec.Mount, error) {
+	newMount := spec.Mount{
+		Type:        define.TypeBind,
+		Source:      "",
+		Destination: "",
+		Options:     append(bindOptions, "ro", "nosuid", "nodev"),
+	}
+
+	tmpDir, err := c.runtime.TmpDir()
+	if err != nil {
+		return newMount, fmt.Errorf("getting runtime temporary directory: %w", err)
+	}
+	tmpDir = filepath.Join(tmpDir, "infra-container")
+	err = os.MkdirAll(tmpDir, 0777)
```

Comment:
when is this ever cleaned up?

Reply:
systemd-tmpfiles could delete this directory after some time.  We need to set the sticky bit on the directory to prevent that

Reply:
is 0777 too broad?

Reply:
I used 0777, because Paul suggested it might be needed, but I have not verified if it actually works fine without that. I will test it out.

For the clean-up, that's good point. I've checked other places in the code where these directories like that are created and haven't seen any clean-up, so I presumed there is some global clean-up done for that c.runtime.TmpDir(). I will read it more to find out how the clean-up actually works there.

Reply:
It seems it works well also with 0755. I still have to ask Paul what he thinks about clean-up.

Reply:
yeah 755 is fine as the overlay should have proper rw permission in the idmappings.

@giuseppe It is a single directory under the libpod tmpdir I don't think we need to clean it up ever. Just like we never cleaned up old pause images.
For the sticky bit thing I assume we already have the set on the libpod tmpdir so all child directories should be safe?

Reply:
ok thanks, yes the sticky bit on the podman tmpdir directory is enough